RoddyCore/src/de/dkfz/roddy/config/validation/DefaultValidator.groovy

```diff
@@ -1,123 +1,123 @@
+
+    boolean result = false
+
+    DefaultValidator(Configuration cfg) {
+        super(cfg)
+    }
+
+    @Override
+    boolean validate(ConfigurationValue configurationValue) {
+        super.errors.clear()
+        super.warnings.clear()
+        EnumerationValue ev = configurationValue.getEnumerationValueType()
+
+        if (configurationValue.value == null) return true
+
+        boolean result = isConfigurationValueTypeCorrect(configurationValue, ev)
+        result &= areDollarCharactersAttached(configurationValue)
+        result &= areVariablesProperlyDefined(configurationValue)
+        this.result = result
+        return result
+    }
+
+    private boolean isConfigurationValueTypeCorrect(ConfigurationValue cv, EnumerationValue ev) {
+        String evID = ev != null ? ev.getId() : CVALUE_TYPE_STRING
+        try {
+            if (evID == CVALUE_TYPE_INTEGER) {
+                cv.toInt()
+            } else if (evID == CVALUE_TYPE_BOOLEAN) {
+                cv.toBoolean()
+            } else if (evID == CVALUE_TYPE_FLOAT) {
+                cv.toFloat()
+            } else if (evID == CVALUE_TYPE_DOUBLE) {
+                cv.toDouble()
+            } else if (evID == CVALUE_TYPE_STRING) {
+            }
+        } catch (Exception e) {
+            super.errors << new ConfigurationIssue(valueAndTypeMismatch, cv.id, cv?.configuration?.file?.toString(), evID)
+            return false
+        }
+        return true
+    }
+
+    /**
+     * Trick: Remove escaped escape characters! This way only single escape characters will be used for the check.
+     */
+    static String removeEscapedEscapeCharacters(String temp) {
+        // Why the loop? Because replaceAll won't do it and will totally replace e.g. odd escape sequence, e.g.:
+        // \\abc => abc     and
+        // \\\abc => abc    as well!
+        while (temp.contains("\\\\"))
+            temp = temp.replace("\\\\", "")
+        temp
+    }
+
+    /** Match either a detached dollar (without escape and brace) somewhere in the text
+     *   OR a non escaped detached dollar at line end.
+     *   OR a detached dollar string '$'.
+     *   OR a detached dollar at the beginning of the line.
+     */
+    static String matchDetachedDollars = /([^\\][$][^{])|([^\\][$]$)|(^[$]$)|(^[$][^{])/
+
+    boolean areDollarCharactersAttached(ConfigurationValue cv) {
+        if (!cv.value.contains('$')) return true
+        if (cv.value == '${$}') return true   // $$ is the current process ID variable, which is fine as value.
+
+        String temp = removeEscapedEscapeCharacters(cv.value)
+
+        boolean result = !temp.findAll(matchDetachedDollars)
+
+        if (!result) {
+            super.warnings << new ConfigurationIssue(detachedDollarCharacter, cv.id, cv?.configuration?.file?.toString())
+        }
+
+        return result
+    }
+
+    static final String VARIABLE = '(^|(?<=[^\\\\{]))[$][{]([\\w[-]_]){1,}[}]'
```

Comment:
This fixes a small bug when matching variable references in configuration values. See the tests in:

https://github.com/TheRoddyWMS/Roddy/pull/357/files?diff=split&w=1#diff-3b522784d7050e976297f3f71b1591edc1c0cd14c4ef5c28f11ff92bc72fb32c